### PR TITLE
VIH-9412 - filter admin web booking list by allocation

### PR DESCRIPTION
**Pull request by marcogagliardi**
https://tools.hmcts.net/jira/browse/VIH-9412

added filters for Justice Users and Not Allocated

### CODE REVIEW

File: AdminWebsite/AdminWebsite/ClientApp/src/app/services/bookings-persist.service.spec.ts [hunk 1]
```diff
@@ -136,6 +136,30 @@ describe('BookingsPersistService', () => {
         });
     });
 
+    describe('#noAllocated', () => {
+        it('should set noAllocatedHearings', () => {
```
Comment (shaed-parkar) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/services/bookings-persist.service.spec.ts:140]: these tests don't offer much other than assert on a value you set yourself. There should be a behaviour change or something (i.e. a function call) between the arrange and assert

**Review by shaed-parkar — COMMENTED** [review]
I think there's room for improvement here. More importantly I think the tests need to offer more than assert a value that has been set but not been tested against some behaviour change else it's just test for coverage

File: AdminWebsite/AdminWebsite/ClientApp/src/app/services/bookings-persist.service.spec.ts (hunk 1 above)
Comment (shaed-parkar) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/services/bookings-persist.service.spec.ts:140]: Is there anything in the `bookings-persist.service.ts` that manipulate these fields that can be called as a part of the test?

File: AdminWebsite/AdminWebsite/Controllers/UserIdentityController.cs [hunk 2]
```diff
@@ -51,5 +52,26 @@ public async Task<ActionResult<UserProfileResponse>> GetUserProfile()
 
             return Ok(profile);
         }
+        
+        [HttpGet("list")]
+        [SwaggerOperation(OperationId = "GetUserList")]
+        [ProducesResponseType(typeof(List<JusticeUserResponse>), (int)HttpStatusCode.OK)]
+        public async Task<ActionResult<ICollection<JusticeUserResponse>>> GetUserList()
```
[the remark this answers is not in the document]
Comment (marcogagliardi) [reply AdminWebsite/AdminWebsite/Controllers/UserIdentityController.cs:59]: done

File: AdminWebsite/AdminWebsite.UnitTests/Controllers/UserIdentityControllerTests.cs [hunk 3]
```diff
@@ -30,8 +32,31 @@ public void Setup()
                 IsVhTeamLeader = true
             };
 
+            _justiceUserListResponse = new List<JusticeUserResponse>();
```
[the remark this answers is not in the document]
Comment (marcogagliardi) [reply AdminWebsite/AdminWebsite.UnitTests/Controllers/UserIdentityControllerTests.cs:35]: done

**Review by will-craig — APPROVED** [review]

File: AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/bookings-list/bookings-list.component.ts [hunk 4]
```diff
@@ -400,6 +415,17 @@ export class BookingsListComponent implements OnInit, OnDestroy {
         );
     }
 
+    private loadUsersList(): void {
+        const self = this;
+        this.videoHearingService.getUsers().subscribe(
+            (data: JusticeUserResponse[]) => {
+                this.users = data;
```
Comment (oliver-scott) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/bookings-list/bookings-list.component.ts:422]: are we showing all of the users in this dropdown list? It should only show the users in the CSO role

Comment (marcogagliardi) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/bookings-list/bookings-list.component.ts:422]: it is not in the acceptance criteria

**Pull request merged**